File: 1288-maximum-subarray-sum-with-one-deletion/maximum-subarray-sum-with-one-deletion.cpp
class Solution {
public:
    int maximumSum(vector<int>& arr) {
        int n=arr.size();
        vector<vector<int>>dp(n,vector<int>(2,INT_MIN));
        
        dp[0][0]=arr[0];
        dp[0][1]=INT_MIN/2;
        for(int i=1;i<n;i++)
        {
            dp[i][0]=max(dp[i-1][0]+arr[i],arr[i]);
            dp[i][1]=max(dp[i-1][1]+arr[i],dp[i-1][0]);
        }

        int m=INT_MIN;
        for(int i=0;i<n;i++)
        {
            m=max(m,*max_element(dp[i].begin(),dp[i].end()));
        }
        return m;

    }
};

File: 1288-maximum-subarray-sum-with-one-deletion/maximum-subarray-sum-with-one-deletion.py
class Solution:
    def maximumSum(self, arr: List[int]) -> int:
        n=len(arr)
        pref=[0]*n
        suff=[0]*n
        if max(arr) < 0:
            return max(arr)
        pref[0]=max(0,arr[0])
        suff[n-1]=max(0,arr[n-1])
        for i in range(1,n):
            pref[i]=max(pref[i-1]+arr[i],arr[i])
        for i in range(n-2,-1,-1):
            suff[i]=max(arr[i],suff[i+1]+arr[i])
        
        mx=-float("inf")
        for i in range(n):
            mx=max(mx,pref[i])
            mx=max(mx,suff[i])
        
        for i in range(1,n-1):
            mx=max(mx,pref[i-1]+suff[i+1])
        
        return mx
        